DOCSP-43666: 2.29 updates (#234)
mongodb/docs-csharp: source/whats-new.txt

.. _upcoming-breaking-changes:

.. _csharp-version-2.29:

What's New in 2.29
------------------

> The 2.29 driver release has been tested against {+mdb-server+} version 8.0 and
> includes the following new features:

Reviewer: S: this phrasing doesn't fully explain whether the version is compatible or partially compatible
maybe something like this is clearer:
```suggestion
The 2.29 driver release is fully compatible with {+mdb-server+} version 8.0 and
includes the following new features:
```

Author: According to the release notes, there are some 8.0 features that aren't compatible. Can update this wording to reflect that.

- Adds support for v2 of the Queryable Encryption range protocol.

> - Adds support for range indexes for Queryable Encryption.

Reviewer: Add a link to more info about QE: see https://www.mongodb.com/docs/drivers/node/current/whats-new/#std-label-version-6.9:~:text=Adds%20official%20support%20for%20Queryable%20Encryption%20(QE)%20range%20queries.%20To%20use%20this%20feature%2C%20your%20app%20must%20use%20mongodb%2Dclient%2Dencryption%20v6.1.0%20or%20later%20and%20must%20connect%20to%20MongoDB%20Server%208.0%20or%20later.%20For%20more%20information%20about%20QE%20range%20queries%2C%20see%20Queryable%20Encryption%20in%20the%20MongoDB%20Server%20manual.
